File: online-judge-solutions/Codeforces/1722C.py
#https://codeforces.com/problemset/problem/1722/C
#Lang		:	Python 3.8
#Time		:	358 ms
#Memory         :	1100 KB
for _ in range(int(input())):
    n=int(input())
    v,m=[[0 for j in range(n)] for i in range(3)],{}
    for i in range(3):
        x=input().split()
        for k in range(len(x)):
            m[x[k]]=m.get(x[k],0)+1
            v[i][k%n]=x[k]
    r=[0 for i in range(3)]
    for i in range(3):
        for j in range(n):
            r[i]+=(3 if m[v[i][j]]==1 else (1 if m[v[i][j]]==2 else 0))
    print(*r)

File: online-judge-solutions/Codeforces/1722C.cpp
//https://codeforces.com/problemset/problem/1722/C
//Lang		:	GNU C++14
//Time		:	156 ms
//Memory	:	400 KB
#include<iostream>
#include<vector>
#include<map>
using namespace std;
#define fastInp cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(0);
int main(){
	fastInp;
	int t,n,p;
	cin>>t;
	while(t--){
		string x;
		cin>>n;p=n*3;
		vector<vector<string>> v(3,vector<string>(n));
		map<string,int> m;
		for(int i=0;i<p;i++){
			cin>>x;m[x]++;
			v[i/n][i%n]=x;
		}
		vector<int> r(3,0);
		for(int i=0;i<v.size();i++){
			for(int j=0;j<v[i].size();j++){
				r[i]+=(m[v[i][j]]==1?3:(m[v[i][j]]==2?1:0));
			}
		}
		for(int z:r){cout<<z<<" ";}cout<<endl;
	}
	return 0;
}
